cluster/get-kube-binaries.sh

```diff
@@ -168,8 +157,8 @@ function download_tarball() {
     # if the url belongs to GCS API we should use oauth2_token in the headers
     curl_headers=""
     if { [[ "${KUBERNETES_PROVIDER:-gce}" == "gce" ]] || [[ "${KUBERNETES_PROVIDER}" == "gke" ]] ; } &&
-       [[ "$url" =~ ^https://storage.googleapis.com.* ]] && valid-storage-scope ; then
```

Comment:
I don't think we ever needed the valid-storage-scope check, if the SA doesn't have scope then we pass no auth, so the resource must be public read ...?

This could (?) bite us later, but only for non-public resources (security patch testing?), if it does I expect to have fun figuring out a follow up with @krzyzacy ;-)

We should not be using the metadata server for this.

Reply:
(if and only if you enable workload identity then the metadata server is part of the pod and it is safe to use)

Otherwise this makes jobs non-portable as they have implicit dependencies on the details of the node the pod gets scheduled on (which we do not want).

Reply:
the previous check asked if the default SA for the metadata server had storage scopes before passing the auth header, if it doesn't it passes no headers.

I'm not sure that makes sense regardless of workload identity. That would mean you would get an error about not being authenticated instead of an error about the auth not having IAM permission.

I'm not sure this ever made sense.